pkg/assets/static/redirectpage.js

```diff
@@ -0,0 +1,189 @@
+  }
+
+  if (link === "notebookController") {
+    resultUrl = `${baseUrl}notebookController/spawner`;
+  } else if (link === "dashboard") {
+    resultUrl = `${baseUrl}dashboard`;
+  } else {
+    resultUrl = redirectUrl.toString();
+  }
+  const url = new URL(resultUrl);
+  Object.keys(params).forEach((key) => {
+    if (params[key]) {
+      url.searchParams.append(key, params[key]);
+    }
+  });
+  window.location.href = url.toString();
+}
+
+function handleError(error) {
+  window.location.href = consoleUrlMock;
+}
+
+function handleUnauthenticated() {
+  idToken = null;
+  window.location.href = consoleUrlMock;
+}
+
+function handleUserInfo() {
+  idToken = keycloak.idToken;
+}
+
+function refreshToken() {
+  keycloak.updateToken(30).catch((error) => {
+    console.error("Failed to refresh token:", error);
+    handleUnauthenticated();
+  });
+}
+
+function initializeKeycloak(clientConfig) {
+  keycloak = new Keycloak(clientConfig);
+  keycloak
+    .init({
+      onLoad: "check-sso",
+      silentCheckSsoRedirectUri:
+        window.location.origin + "/silent-check-sso.html",
+    })
+    .then((authenticated) => {
+      if (authenticated) {
+        console.log("User is authenticated");
+        intervalRefRefresh = setInterval(refreshToken, 15000); //start 15s interval token refresh
+        keycloak
+          .loadUserInfo()
+          .then(handleUserInfo)
+          .catch(() => handleError("failed to pull in user data."));
+      } else {
+        console.log("user not authenticated");
+        handleUnauthenticated();
+      }
+    })
+    .catch(() => handleError("Failed to initialize authorization"));
+}
+
+// General function to fetch JSON data
+function getJSON(
+  method,
+  url,
+  token,
+  callback,
+  body = null,
+  headers = {}
+) {
+  let xhr = new XMLHttpRequest();
+  xhr.open(method, url, true);
+  if (token != null) xhr.setRequestHeader("Authorization", "Bearer " + token);
+
+  Object.entries(headers).forEach(([key, value]) => {
+    xhr.setRequestHeader(key, value);
+  });
+
+  xhr.responseType = "json";
+  xhr.onload = () => {
+    let status = xhr.status;
+    if (status >= 200 && status < 300) {
+      callback(null, xhr.response);
+    } else {
+      callback(status, xhr.response);
+    }
+  };
+  xhr.send(body ? JSON.stringify(body) : null);
+}
+
+function loadAuthLibrary(url, cbSuccess, cbError) {
+  const script = document.createElement("script");
+  script.setAttribute("src", url);
+  script.setAttribute("type", "text/javascript");
+  let loaded = false;
+  function loadFunction() {
+    if (loaded) return;
+    loaded = true;
+    cbSuccess();
+  }
+  function errorFunction(error) {
+    if (loaded) return;
+    cbError(error);
+  }
+  script.onerror = errorFunction;
+  script.onload = loadFunction;
+  script.onreadystatechange = loadFunction;
+  document.head.appendChild(script);
+}
+
+async function getRedirectData() {
+  const xhr = new XMLHttpRequest();
+
+  xhr.open("GET", registrationURL, true);
```

Comment:
Hm... I don't actually see in my browser any requests to `"/api/v1/signup"`. This is what I see instead:
- redirectpage.html redirects to sso (expected)
- then it redirects to console.openshift.com/sandbox even if I already have an account in dev sandobx. So it doesn't seem to actually check `/api/v1/signup` unless I'm missing something. But anyway, no matter what I do I always end up in console.openshift.com